quick-start/demo-function/README.md

```diff
@@ -0,0 +1,43 @@
+# demo-function
+
+This directory contains a demo function for the purposes of testing the
+vault-lambda-extension. It attempts to connect and query users from a database
+using two different sets of credentials; one set fetched from disk (which the
+extension wrote during initialization) and the other set from the proxy server
+which is queried during the function runtime.
+
+The `quick-start` guide defaults to deploying this function in a zip format
+Lambda function. See the readme in the parent `quick-start` folder for full
+instructions on deploying it.
+
+Alternatively, to deploy the `quick-start` guide using a container format Lambda,
+make the following modifications:
+
+* Create an ECR repository. You can use AWS CLI or console, or add this to
+  the Terraform config:
+
+  ```hcl
+  resource "aws_ecr_repository" "demo-function" {
+    name                 = "demo-function"
+  }
+  ```
+
+* Package the extension and function together into a Docker image. From this
+  directory:
+
+  ```bash
+  pushd ../../ && GOOS=linux GOARCH=amd64 go build -ldflags '-s -w' -a -o quick-start/demo-function/pkg/extensions/vault-lambda-extension main.go && popd
+  docker build -t demo-function .
```

Comment:
Wondering if we should just have a Makefile and do something like `make build`?

Reply:
I didn't want to screw around with the well-tested `build.sh` script too much, and would have wanted to fold it into the Makefile, so instead I added a `build-container.sh` script alongside `build.sh`. That keeps the instructions much simpler.